src/cmd/policy.go

```diff
@@ -136,6 +144,61 @@ func RegoFuncGetGithubRepo(ctx rego.BuiltinContext, a, b *ast.Term) (*ast.Term,
 
 	reader := strings.NewReader(response.String())
 	v, err := ast.ValueFromReader(reader)
+
+	return ast.NewTerm(v), nil
+}
+
+func RegoFuncGetGitlabRepo(ctx rego.BuiltinContext, a *ast.Term) (*ast.Term, error) {
+
+	var path string
+	if err := ast.As(a.Value, &path); err != nil {
+		return nil, err
+	}
+
+	if path == "" {
+		log.Error().Msgf("opslevel.repo.gitlab(\"%s\") failed: Please provide a valid Gitlab project path", path)
+		return nil, nil
+	}
+
+	escapedPath := url.QueryEscape(path)
+	gitlabToken := viper.GetString("gitlab-token")
+	gitlabAPIUrl := fmt.Sprintf("https://gitlab.com/api/v4/projects/%s", escapedPath)
+
+	type GitlabRepoMetadata struct {
+		Name        string             `json:"name,omitempty"`
+		Description string             `json:"description,omitempty"`
+		Languages   map[string]float32 `json:"languages,omitempty"`
```

Comment:
is the data format of this different from the Github one?  We should keep them insync in terms of data format exposed to the policy.

Reply:
Yeah, we're just getting top language on the Github side and Gitlab doesn't return that so I made the extra call to get Languages which is similar to what Github would do.  Not sure what the Github one returns for each language though, lines? Either way it uses these numbers to calculate a percentage which I assume we would want to do when we start pulling in languages?  Maybe I should add a Language field similar to what Github and set it equal to the highest percentage?
```
{
  "Go": 104748,
  "Shell": 1335,
  "Dockerfile": 379
}
```

Reply:
Just so that I'm on the same page, are you saying that we should have the same fields returned for each function?  Or as close as possible?  Right now Github is returning everything so I agree that it's very different UX for each function as it is.  Should I go ahead and make this struct repo metadata and use it for both the github and gitlab functions? That way it returns the same info?